pkg/token/federatedIdentity.go

```diff
@@ -114,3 +111,54 @@ func readJWTFromFS(tokenFilePath string) (string, error) {
 	}
 	return string(token), nil
 }
+
+func getGitHubToken() (string, error) {
```

Comment:
let's pass the context into this function since it's doing an external http call